src/api/base.ts

```diff
@@ -0,0 +1,39 @@
+const baseUrl = `http://localhost:8000/api/`;
+
+const apiUrl = (endpoint: string) => {
+    return `${baseUrl}${endpoint}`;
+};
+
+export const fetchToken = async (data: { username: string, password: string }) => {
+    const response = await fetch(apiUrl(`token/`), {
+        method: `POST`,
+        headers: {
+            "Content-Type": `application/json`
+        },
+        body: JSON.stringify(data)
+    });
+    const token = await response.json();
+
+    if (token.access) {
+        localStorage.setItem(`token`, token.access);
+    }
+    return token;
+};
+
+export const fetchFunds = async () => {
+    const response = await fetch(apiUrl(`funds/`), {
+        method: `GET`,
+        headers: {
+            "Content-Type": `application/json`,
+            "Authorization": `Bearer ${localStorage.getItem(`token`)}`,
+        }
+    });
+
+    const json = await response.json();
+
+    if (json.detail) {
```

Comment:
`detail` is only present on error. Probably better to do some sort of error handling and raising custom api errors that are themselves handled downstream. Not worth the effort right now, but making issue for it now.

Reply:
https://github.com/ksiondag/base_webapp/issues/9